src/Middleware/HttpsPolicy/test/HstsMiddlewareTests.cs

```diff
@@ -15,13 +15,152 @@
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Logging.Testing;
+using Microsoft.Extensions.Options;
 using Microsoft.Net.Http.Headers;
 using Xunit;
 
 namespace Microsoft.AspNetCore.HttpsPolicy.Tests
 {
     public class HstsMiddlewareTests
     {
+#region Unit tests
+        [Fact]
+        public void HstsMiddleware_ArgumentNextIsNull_ThrowsArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                _ = new HstsMiddleware(next: null, options: new OptionsWrapper<HstsOptions>(new HstsOptions()));
+            });
+        }
+
+        [Fact]
+        public void HstsMiddleware_ArgumentOptionsIsNull_ThrowsArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                _ = new HstsMiddleware(innerHttpContext => Task.CompletedTask, options: null);
+            });
+        }
+
+        [Fact]
+        public async Task Invoke_SetsHstsHeader()
+        {
+            var middleware = new HstsMiddleware(innerHttpContext => Task.CompletedTask, new OptionsWrapper<HstsOptions>(new HstsOptions()));
+
+            var context = new DefaultHttpContext();
+            context.Request.Host = new HostString("example.com");
+            context.Request.Scheme = "https";
+
+            await using (context.Response.Body = new MemoryStream())
```

Comment:
Why are you dealing with the response body in these tests? The middleware is unrelated to the response body.

Reply:
Fixed